include/ck/utility/amd_buffer_addressing.hpp

```diff
@@ -944,4 +944,33 @@ amd_buffer_atomic_max(const typename vector_type_maker<T, N>::type::type src_thr
 #endif
 }
 
+// Direct loads from global to LDS.
+__device__ void
+llvm_amdgcn_raw_buffer_load_lds(int32x4_t rsrc,
+                                __attribute__((address_space(3))) float* lds_ptr,
+                                index_t size,
+                                index_t voffset,
+                                index_t soffset,
+                                index_t offset,
+                                index_t aux) __asm("llvm.amdgcn.raw.buffer.load.lds");
+
+template <typename T>
+__device__ void amd_direct_load_global_to_lds(const T* global_base_ptr,
+                                              const index_t global_offset,
+                                              void* lds_base_ptr,
+                                              const index_t lds_offset)
+{
+    // Direct loads require that each thread writes a single DWORD.
+    static_assert(sizeof(T) == 4);
+
+    const int32x4_t src_resource = make_wave_buffer_resource_with_default_range(global_base_ptr);
```

Comment:
Why use make_wave_buffer_resource_with_default_range instead of make_wave_buffer_resource with element_wise_size?

Reply:
Sure, created a new `CopyTo` method that takes another another buffer as the argument and copies data directly to it.